keras/preprocessing/sequence.py

```diff
@@ -4,31 +4,39 @@
 import random
 from six.moves import range
 
-def pad_sequences(sequences, maxlen=None, dtype='int32', padding='pre'):
+
+def pad_sequences(sequences, maxlen=None, dtype='int32', padding='pre', vector_sequence=False):
     """
         Pad each sequence to the same length: 
         the length of the longuest sequence.
 
         If maxlen is provided, any sequence longer
-        than maxlen is truncated to maxlen.
+        then maxlen is truncated to maxlen.
+        
+        If vector_sequence is true, then each sequence of numpy
+        array is padded using zero arrays.
 
         Support post-padding and pre-padding (default).
     """
     lengths = [len(s) for s in sequences]
-
     nb_samples = len(sequences)
     if maxlen is None:
         maxlen = np.max(lengths)
-
-    x = np.zeros((nb_samples, maxlen)).astype(dtype)
+    
+    if not vector_sequence:
+        x = np.zeros((nb_samples, maxlen)).astype(dtype)
+    else:
+        dim = len(sequences[0][0])
```

Comment:
What about the edge case where sequences is empty, or (more likely) a list where the first array is empty?